src/caja-window-slot.c

```diff
@@ -384,34 +384,43 @@ caja_window_slot_update_icon (CajaWindowSlot *slot)
     icon_name = NULL;
     if (info)
     {
+        GtkWindow *gtk_window;
+        GdkPixbuf *pixbuf;
+
+        gtk_window = GTK_WINDOW (window);
+
+        pixbuf = gtk_window_get_icon (gtk_window);
+        if (pixbuf)
+            gtk_window_set_icon (gtk_window, NULL);
+
         icon_name = caja_icon_info_get_used_name (info);
         if (icon_name != NULL)
         {
             /* Gtk+ doesn't short circuit this (yet), so avoid lots of work
              * if we're setting to the same icon. This happens a lot e.g. when
              * the trash directory changes due to the file count changing.
              */
-            if (g_strcmp0 (icon_name, gtk_window_get_icon_name (GTK_WINDOW (window))) != 0)
+            const gchar *current_icon_name;
+
+            current_icon_name = gtk_window_get_icon_name (gtk_window);
+            if ((current_icon_name == NULL) || (strcmp (icon_name, current_icon_name) != 0))
             {
-                if (g_strcmp0 (icon_name, "text-x-generic") == 0)
-                    gtk_window_set_icon_name (GTK_WINDOW (window), "folder-saved-search");
+                if (strcmp (icon_name, "text-x-generic") == 0)
```

Comment:
the same here

Reply:
`icon_name` can't be NULL... take a look at the line 397 `if (icon_name != NULL)`